R/tidy-upkeep.R

```diff
@@ -30,16 +68,67 @@ use_tidy_upkeep_issue <- function(year = NULL) {
   view_url(issue$html_url)
 }
 
-upkeep_checklist <- function(year = NULL,
-                             posit_pkg = is_posit_pkg(),
-                             posit_person_ok = is_posit_person_canonical()) {
-  year <- year %||% 2000
+upkeep_checklist <- function(year = NULL) {
+
+  bullets <- c(
+    year,
+    "",
+    todo("`usethis::use_readme_rmd()`", !file_exists("README.Rmd")),
+    todo("`usethis::use_roxygen_md()`", !is_true(uses_roxygen_md())),
+    todo("`usethis::use_github_links()`", !has_github_links()),
+    todo("`usethis::use_pkgdown_github_pages()`", !uses_pkgdown()),
+    todo("`urlchecker::url_check()`"),
+    todo(
+      "
+        `usethis::use_package_doc()`
+        Consider letting usethis manage your `@importFrom` directives here.
+        `usethis::use_import_from()` is handy for this.",
+      !has_package_doc()
+    ),
+    todo(
+      "
+        `usethis::use_testthat(3)` and upgrade to 3e, \\
+        [testthat 3e vignette](https://testthat.r-lib.org/articles/third-edition.html)",
+      !uses_testthat_edition(3)
+    ),
+    todo("
+        Align the names of `R/` files and `test/` files for workflow happiness.
+        The docs for `usethis::use_r()` include a helpful script.
+        `usethis::rename_files()` may be be useful."),
+    todo("usethis::use_github_action('check-standard')", !uses_github_actions()),
+    todo(
+      "Consider changing default branch from `master` to `main`",
+      git_default_branch() == "master"
```

Comment:
I'm wondering if the machinations required in the tests to make this work are a code smell, and we should just put this bullet in unconditionally?

Reply:
Below I suggest using some mocks in the tests.